### PR TITLE
DATA: Configuration Edits

### LINKED ISSUE #141: DATA: Configuration
```cs
  public class Configuration
  {
      public Guid Id { get; set; }
      public string Name { get; set; }
      public string Value { get; set; }
      public string Type { get; set; } 
      public string CreatedBy { get; set; }
      public DateTimeOffset CreatedDate { get; set; }
      public string UpdatedBy { get; set; }
      public DateTimeOffset UpdatedDate { get; set; }
  }
```
**Required properties:**
- Name
- Value
- Type

**Unique properties:**
- Name

### PR DESCRIPTION
**Pull request by makhammadsoliyev**
Closes #141 

Add Configuration Model and Migration

### CODE REVIEW

**Comment by glhays** [comment]
@makhammadsoliyev please fix your Title block here. example=>DATA: configuration Edits

**Comment by makhammadsoliyev** [comment]
> @makhammadsoliyev please fix your Title block here. example=>DATA: Configuration Edits

solved

**Comment by glhays** [comment]
> > @makhammadsoliyev please fix your Title block here. example=>DATA: Configuration Edits
> 
> solved

Thank you.

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.Configurations.Configuration.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using GitFyle.Core.Api.Models.Foundations.Configurations;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace GitFyle.Core.Api.Brokers.Storages
+{
+    internal partial class StorageBroker
+    {
+        private void AddConfigurationConfigurations(EntityTypeBuilder<Configuration> builder)
+        {
+            builder
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Storages/StorageBroker.Configurations.Configuration.cs:14]: Can you remove this restriction of 255 on Name Value and Type.

Comment (makhammadsoliyev) [reply GitFyle.Core.Api/Brokers/Storages/StorageBroker.Configurations.Configuration.cs:14]: I did as you asked

File: GitFyle.Core.Api/Migrations/20240730193424_AddConfigurations.Designer.cs [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+                        .HasColumnType("datetime2");
+
+                    b.Property<string>("Name")
+                        .IsRequired()
+                        .HasMaxLength(255)
+                        .HasColumnType("nvarchar(255)");
+
+                    b.Property<string>("Type")
+                        .IsRequired()
+                        .HasMaxLength(255)
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Migrations/20240730193424_AddConfigurations.Designer.cs:47]: You need to redo the migration as the length restriction is still on the migration

Comment (makhammadsoliyev) [reply GitFyle.Core.Api/Migrations/20240730193424_AddConfigurations.Designer.cs]: Done

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**